### DISCUSSION TITLE
Is it possible to add a command before bash in nxf_trace() of .command.run?

### DISCUSSION THREAD

**Discussion by lishengting** [nextflow-io/nextflow#4846] [Q&A] [answered]
![image](https://github.com/nextflow-io/nextflow/assets/10738140/7cb3b9b6-052f-478c-a9c2-87d4a7152f28)
Just as showed in above picture.
Is it possible to add /path/to/stat_tool before /bin/bash，or even replace /bin/bash by another software written by myself?

Thanks!

**Comment by robsyme** [comment, marked as answer]
Absolutely! There are in fact a couple of ways. 

You might consider adding it in the shebang line of the script block.

```groovy
process Demo {
    script:
    """
    #!/path/to/stat_tool
    echo 'Hello world!'
    """
}
```

Nextflow will detect the shebang line and adjust the `.command.run` statement accordingly:

```bash
nxf_launch() {
   /path/to/stat_tool /private/tmp/lishengting/work/a7/9f73aef2239268b6a44213553372cc/.command.sh
}
```

You can also adjust the interpreter with the `shell` process directive ([documentation here](https://www.nextflow.io/docs/latest/process.html#process-directive-shell)):

```groovy
process Demo {
    debug true
    shell '/path/to/stat_tool', '--extra-args', 'if_necessary'

    script:
    "echo 'Hello world!'"
}
```

Of course, because this is a process directive it can also be overridden via configuration. You might have a `nextflow.config` file that includes:

```groovy
process {
  withName: Demo {
    shell = '/path/to/stat_tool'
  }
}
```

**Reply by lishengting** [reply]
That's very convenient.
Thanks a lot!
